src/ServiceClient.php

```diff
@@ -44,15 +44,49 @@ public function setProtocol($protocol)
 
     public function get($path, $query = null)
     {
+        $uri = $this->createUri($path, $query);
+
+        $result = $this->guzzleClient->get($uri);
+
+        return json_decode(strval($result->getBody()));
+    }
+
+    public function delete($path, $query = null)
+    {
+        $uri = $this->createUri($path, $query);
+
+        $result = $this->guzzleClient->delete($uri);
+
+        return json_decode(strval($result->getBody()));
+    }
+
+    public function put($path, $query = null, $payload = null)
+    {
+        $uri = $this->createUri($path, $query);
+
+        $result = $this->guzzleClient->put($uri, $payload);
+
+        return json_decode(strval($result->getBody()));
+    }
+    
+    public function post($path, $query = null, $payload = null)
+    {
+        $uri = $this->createUri($path, $query);
+
+        $result = $this->guzzleClient->post($uri, $payload);
```

Comment:
I think guzzle expects more structure on the second argument -- the `post` method suggests that `$payload` *is* the payload of the POST, but I think for guzzle you need to actually send `[ 'body' => $payload ]` (this is based on a quick review of the docs -- probably best to verify with a quick test in case I am misreading).

It might feel less repetitious to centralize the request / response handling and just use the verb-specific methods to determine one or two of the parameters to a generic `makeRequest` method, e.g.:
```php
private function makeRequest($method, $path, $query = null, $payload = null) {
  $baseUri = $this->serviceDiscoveryClient->getServiceAddress($this->serviceName, $this->serviceVersion);
  
  $uri = "{$this->protocol}://$baseUri/$path";
  if (!empty($query)) {
    $uri .= "?" . urlencode($query);
  }

  $options = [];  
  if (!empty($payload)) {
    $options['body'] = $payload;
  }

  $result = $this->guzzleClient->request($method, $uri, $options);
  
  return json_decode(strval($result->getBody()));
}

public function post($path, $query = null, $payload = null) {
  return $this->makeRequest('POST', $path, $query, $payload);
}
```